### PR TITLE
HPCC-25498 Check and send error messages in DFUFileAccessV2

### PR DESCRIPTION
**Pull request by wangkx**
1. Check the keyPairName setting in /EnvSettings/Keys/ClusterGroup.
2. Check the privateKey setting in /EnvSettings/Keys/KeyPair.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/clients/wsdfuaccess/wsdfuaccess.cpp [hunk 1]
```diff
@@ -525,6 +525,8 @@ StringBuffer &encodeDFUFileMeta(StringBuffer &metaInfoBlob, IPropertyTree *metaI
 
         const char *keyPairName = metaInfo->queryProp("keyPairName");
         const char *privateKeyFName = environment->getPrivateKeyPath(keyPairName);
+        if (isEmptyString(privateKeyFName))
+            throw makeStringExceptionV(-1, "Key name %s is not found in environment settings: /EnvSettings/Keys/KeyPair.", keyPairName);
```
Comment (jakesmith) [inline esp/clients/wsdfuaccess/wsdfuaccess.cpp:529]: trivial: perhaps change to '%s' - it can sometimes be confusing depending on naming if unquoted.

**Review by jakesmith — COMMENTED** [review]
@wangkx - looks fine, 1 trivial issue, please address and squash.

File: esp/clients/wsdfuaccess/wsdfuaccess.cpp (hunk 1 above)
Comment (wangkx) [reply esp/clients/wsdfuaccess/wsdfuaccess.cpp]: @jake I just made the change.

File: esp/services/ws_dfu/ws_dfuService.cpp [hunk 2]
```diff
@@ -6030,6 +6030,10 @@ void CWsDfuEx::dFUFileAccessCommon(IEspContext &context, const CDfsLogicalFileNa
     unsigned port;
     bool secure;
     getFileDafilesrvConfiguration(keyPairName, port, secure, fileName, groups);
+#ifdef _USE_OPENSSL
+    if (secure && keyPairName.isEmpty())
+        throw makeStringExceptionV(-1, "No keyPairName is found for %s in environment settings: /EnvSettings/Keys/ClusterGroup.", cluster.str());
```
Comment (jakesmith) [inline esp/services/ws_dfu/ws_dfuService.cpp:6035]: @wangkx - here also

Comment (wangkx) [reply esp/services/ws_dfu/ws_dfuService.cpp]: @jakesmith, now it is fixed.

**Review by jakesmith — DISMISSED** [review]
@wangkx - looks good

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Comment by ghalliday** [comment]
@wangkx the PR now contains an invalid update to the aws submodule

**Comment by wangkx** [comment]
@ghalliday not sure why I cannot see the invalid update in the PR. I will clean it.

**Review by jakesmith — COMMENTED** [review]
@wangkx - will approve again when aws submodule issue resolved

**Comment by jakesmith** [comment]
> @ghalliday not sure why I cannot see the invalid update in the PR. I will clean it.

image

**Comment by wangkx** [comment]
@jakesmith the aws submodule issue is resolved

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**